libs/execution_context/ebpf_maps.c

```diff
@@ -197,6 +206,24 @@ _delete_array_map_with_references(_In_ ebpf_core_map_t* map)
     ebpf_free(map);
 }
 
+static ebpf_result_t
+_associate_program_with_prog_array_map(_In_ ebpf_core_map_t* map, _In_ const ebpf_program_t* program)
+{
+    ebpf_assert(map->ebpf_map_definition.type == BPF_MAP_TYPE_PROG_ARRAY);
+    ebpf_program_array_map_t* program_array = (ebpf_program_array_map_t*)map;
+
+    // Validate that the program type is
+    // not in conflict with the map's program type.
+    const ebpf_program_type_t* program_type = ebpf_program_type(program);
+    if (program_array->program_type == NULL) {
+        program_array->program_type = program_type;
```

Comment:
We are saving the pointer to program type (which is owned by the ebpf_program_t struct) in ebpf_program_array_map_t. It is possible that the map outlives the program (map is pinned, program is unloaded), then this pointer will be invalid as ebpf_program_t structure will be freed?